### ISSUE TITLE
Using StartInvData with BinaryInvestment leads to infeasibility

### ISSUE THREAD

**Issue by dqpinel** [EnergyModelsX/EnergyModelsInvestments.jl#35]
When using the investment type `BinaryInvestment` with the investment data type `StartInvData`, the model becomes infeasible. 
![bilde](https://github.com/user-attachments/assets/300869c5-de86-4491-9ad2-b009bc270b93)

The two types should be made incompatible.

**Comment by JulStraus** [comment]
I generally agree with the comment. The major issue is if we specify a capacity beforehand that is not compatible with the provided capacity increases. This problematic change came up due to the changes in #23 in which we decided to move the capacity used for `BinaryInvestment` from the chosen capacity. In practice, we have the following issues from *[model.jl](https://github.com/EnergyModelsX/EnergyModelsInvestments.jl/blob/e01b1de30e688012feeb083587337964b22a6b4d/src/model.jl)* (link to current main commit):

L.56-57:

```julia
start_cap_val = start_cap(element, t_inv, inv_data, cap)
@constraint(m, var_current[t_inv] == start_cap_val + var_add[t_inv])
```

L. 87 ff:

```julia
@constraint(
    m,
    [t_inv ∈ 𝒯ᴵⁿᵛ],
    var_current[t_inv] ==
    invest_capacity(inv_mode, t_inv) * var_invest_b[element, t_inv]
)
```

In practice, we cannot allow with the current design for `BinaryInvestment` any existing capacity, be it through `StartInvData` (with the specific field) or `NoStartInvData` (through the function `start_cap`). Personally speaking, I would say it is simpler to just allow it for `StartInvData` in which we incorporate an inner constructor to check that the value for `initial` is 0 and forbid it for `NoStartInvData` as we cannot control the given value there.

 I personally see that the solution should come in conjunction with a solution to #30 .

**Comment by JulStraus** [comment]
Solved by #38
